src/lib/genesis_ledger/genesis_ledger.ml

```diff
@@ -99,28 +107,57 @@ include Utils
 
 module Make (Inputs : Intf.Ledger_input_intf) : Intf.S = struct
   include Inputs
+  include Backing
 
   (* TODO: #1488 compute this at compile time instead of lazily *)
-  let t =
+  let backing_ledger =
     let open Lazy.Let_syntax in
     let%map ledger, insert_accounts =
       match directory with
       | `Ephemeral ->
-          lazy (Ledger.create_ephemeral ~depth (), true)
+          lazy (`Ephemeral (Ledger.create_ephemeral ~depth ()), true)
       | `New ->
-          lazy (Ledger.create ~depth (), true)
+          lazy (`Root (Ledger.Root.create_single ~depth ()), true)
       | `Path directory_name ->
-          lazy (Ledger.create ~directory_name ~depth (), false)
+          lazy
+            (`Root (Ledger.Root.create_single ~directory_name ~depth ()), false)
+    in
+    let masked =
+      match ledger with
+      | `Ephemeral ledger ->
+          ledger
+      | `Root ledger ->
+          Ledger.Root.as_masked ledger
     in
     ( if insert_accounts then
       let addrs_and_accounts =
-        let ledger_depth = Ledger.depth ledger in
+        let ledger_depth = Ledger.depth masked in
         Lazy.force accounts
         |> List.mapi ~f:(fun i (_, acct) ->
                (Ledger.Addr.of_int_exn ~ledger_depth i, acct) )
       in
-      Ledger.set_batch_accounts ledger addrs_and_accounts ) ;
-    ledger
+      Ledger.set_batch_accounts masked addrs_and_accounts ) ;
+    (ledger, masked)
+
+  let t = Lazy.map ~f:snd backing_ledger
+
+  let populate_root root =
+    let backing_ledger, mask = Lazy.force backing_ledger in
+    Ledger.commit mask ;
```

Comment:
Some `super_catchup` unit tests were failing, and I believe this is the reason - I'm attempting to checkpoint the `Ledger.t` mask regardless of the underlying ledger. When this is `Ephemeral`, the root is a null ledger, so this is blowing up.